src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/TensorPrimitives.netcore.cs

```diff
@@ -0,0 +1,1062 @@
+        /// <exception cref="ArgumentException">Destination is too short.</exception>
+        /// <remarks>
+        ///     <para>This method effectively does <c><paramref name="destination" />[i] = (<paramref name="x" />[i] * <paramref name="y" />[i]) + <paramref name="addend" /></c>.</para>
+        ///     <para>This method corresponds to the <c>axpy</c> method defined by <c>BLAS1</c>.</para>
+        /// </remarks>
+        public static void MultiplyAdd(ReadOnlySpan<float> x, ReadOnlySpan<float> y, float addend, Span<float> destination) =>
+            InvokeSpanSpanScalarIntoSpan<MultiplyAddOperator>(x, y, addend, destination);
+
+        /// <summary>Computes the element-wise result of: <c>(<paramref name="x" /> * <paramref name="y" />) + <paramref name="addend" /></c>.</summary>
+        /// <param name="x">The first tensor, represented as a span.</param>
+        /// <param name="y">The second tensor, represented as a span.</param>
+        /// <param name="addend">The third tensor, represented as a span.</param>
+        /// <param name="destination">The destination tensor, represented as a span.</param>
+        /// <exception cref="ArgumentException">Length of '<paramref name="x" />' must be same as length of '<paramref name="addend" />'.</exception>
+        /// <exception cref="ArgumentException">Destination is too short.</exception>
+        /// <remarks>This method effectively does <c><paramref name="destination" />[i] = (<paramref name="x" />[i] * <paramref name="y" />) + <paramref name="addend" />[i]</c>.</remarks>
+        public static void MultiplyAdd(ReadOnlySpan<float> x, float y, ReadOnlySpan<float> addend, Span<float> destination) =>
+            InvokeSpanScalarSpanIntoSpan<MultiplyAddOperator>(x, y, addend, destination);
+
+        private static unsafe void InvokeSpanIntoSpan<TUnaryOperator>(
+            ReadOnlySpan<float> x, Span<float> destination)
+            where TUnaryOperator : IUnaryOperator
+        {
+            if (x.Length > destination.Length)
+            {
+                ThrowHelper.ThrowArgument_DestinationTooShort();
+            }
+
+            ref float xRef = ref MemoryMarshal.GetReference(x);
+            ref float dRef = ref MemoryMarshal.GetReference(destination);
+            int i = 0, oneVectorFromEnd;
+
+#if NET8_0_OR_GREATER
+            if (Vector512.IsHardwareAccelerated)
+            {
+                oneVectorFromEnd = x.Length - Vector512<float>.Count;
+                if (i <= oneVectorFromEnd)
+                {
+                    // Loop handling one vector at a time.
+                    do
+                    {
+                        Vector512.StoreUnsafe(
+                            TUnaryOperator.Invoke(
+                                Vector512.LoadUnsafe(ref xRef, (uint)i)),
+                            ref dRef, (uint)i);
```

Comment:
Just wondering, why this way rather than something like:
```csharp
Vector512<float> result = TUnaryOperator.Invoke(Vector512.LoadUnsafe(ref xRef, (uint)i));
result.StoreUnsafe(ref dRef, (uint)i);
```

or
```csharp
TUnaryOperator.Invoke(Vector512.LoadUnsafe(ref xRef, (uint)i))
              .StoreUnsafe(ref dRef, (uint)i);
```

I personally find them a little bit more readable than the nesting/indention here (but not a big deal by any means)

Reply:
> Just wondering, why this way

I was mainly trying to be consistent with the other methods that had more arguments, e.g. three LoadUnsafe calls as arguments to Invoke. I also didn't realize there was an instance StoreUnsafe method.

I'll try to find a happy middle ground.